server/src/test/java/io/crate/execution/ddl/tables/CreateTableRequestTest.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Map;
+
+public class CreateTableRequestTest {
+
+    @Test
+    public void test_streaming() throws Exception {
+        RelationName rel = RelationName.of(QualifiedName.of("t1"), Schemas.DOC_SCHEMA_NAME);
+
+        Reference ref1 = new SimpleReference(
+            new ReferenceIdent(rel, "part_col_1"),
+            RowGranularity.DOC,
+            DataTypes.STRING,
+            1,
+            null,
+            null
+        );
+        Reference ref2 = new SimpleReference(
+            new ReferenceIdent(rel, "part_col_2"),
+            RowGranularity.DOC,
+            DataTypes.INTEGER,
+            2,
+            null,
+            null
+        );
+        Reference ref3 = new SimpleReference(
+            new ReferenceIdent(rel, "just_col"),
+            RowGranularity.DOC,
+            DataTypes.BYTE,
+            3,
+            null,
+            null
+        );
+        Reference ref4 = new SimpleReference(
+            new ReferenceIdent(rel, "some_routing_col"),
+            RowGranularity.DOC,
+            DataTypes.LONG,
+            4,
+            null,
+            null
+        );
+        List<Reference> refs = List.of(ref1, ref2, ref3, ref4);
+        List<String> partCol1 = List.of("part_col_1", typeNameForESMapping(DataTypes.STRING, "english", false));
+        List<String> partCol2 = List.of("part_col_2", typeNameForESMapping(DataTypes.INTEGER, null, false));
+        List<List<String>> partitionedBy = List.of(partCol1, partCol2);
+
+        CreateTableRequest request = new CreateTableRequest(
+            rel,
+            refs,
+            IntArrayList.from(3),
+            Map.of("check1", "just_col > 0"),
+            Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, Version.V_5_4_0).build(),
+            "some_routing_col",
+            ColumnPolicy.DYNAMIC,
+            partitionedBy,
+            Map.of("fulltext_index_name", Map.of())
+        );
+
+        BytesStreamOutput out = new BytesStreamOutput();
+        request.writeTo(out);
+        CreateTableRequest fromStream = new CreateTableRequest(out.bytes().streamInput());
+
+        assertThat(fromStream.getTableName()).isEqualTo(rel);
+        assertThat(fromStream.references()).containsExactlyElementsOf(refs);
+        assertThat(fromStream.checkConstraints()).isEqualTo(Map.of("check1", "just_col > 0"));
+        assertThat(fromStream.pKeyIndices().size()).isEqualTo(1);
+        assertThat(fromStream.pKeyIndices().get(0)).isEqualTo(3);
+
+        assertThat(fromStream.settings()).isEqualTo(Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, Version.V_5_4_0).build());
+        assertThat(fromStream.routingColumn()).isEqualTo("some_routing_col");
+        assertThat(fromStream.tableColumnPolicy()).isEqualTo(ColumnPolicy.DYNAMIC);
+        assertThat(fromStream.partitionedBy()).containsExactlyElementsOf(partitionedBy);
+        assertThat(fromStream.indices().keySet()).containsExactly("fulltext_index_name");
```

Comment:
minor: Please compare the values of the new read request with the values of the written request instead of using concrete values to avoid any issues also later on when doing some possible refactoring.